### PR TITLE
test(dm): inject error to a not-auto-resume source

### LINKED ISSUE #6618: query-status test execute .* error count: 0 != expected: 1
### Which jobs are flaking?

shardddl2

### Which test(s) are flaking?

DM-IT

### Jenkins logs or GitHub Actions link

https://ci2.pingcap.net/blue/organizations/jenkins/dm_ghpr_integration_test/detail/dm_ghpr_integration_test/8612/pipeline

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close #6618

### What is changed and how it works?

mysql-replica-2 has turned off auto-resume, we use its worker to inject failpoint

#### Tests 

 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: dm/tests/shardddl2/run.sh [hunk 1]
```diff
@@ -308,7 +308,7 @@ function DM_DROP_COLUMN_EXEC_ERROR() {
 function DM_DROP_COLUMN_ALL_DONE_CASE() {
 	# get worker of source1
```
Comment (lance6716) [inline dm/tests/shardddl2/run.sh:309]:
```suggestion
	# get worker of source2
```

**Review by buchuitoudegou — APPROVED** [review]

**Review by lichunzhu — CHANGES_REQUESTED** [review]
LGTM

**Review by lichunzhu — APPROVED** [review]
LGTM

**Comment by lichunzhu** [comment]
@buchuitoudegou Sorry I errorly choose `Request for change`. Could you please help approve again?

**Pull request merged**